packages/manager/apps/pci-rancher/src/utils/rancher.ts

```diff
@@ -69,3 +75,88 @@ export const getRancherPlanDescription = (rancherPlan: RancherPlan['name']) => {
       return null;
   }
 };
+
+/**
+ * Extracts drivers and plan information from a switch plan error message.
+ *
+ * @param {string} inputString - The input error message string.
+ * @returns {null|{drivers: string[], plan: string}} - An object containing the drivers and plan, or null if extraction fails.
+ * */
+export function extractDriversAndPlanFromSwitchPlanError(
+  inputString: string,
+): null | { drivers: string[]; plan: string } {
+  const openBracketIndex = inputString.indexOf('[');
+  const closeBracketIndex = inputString.indexOf(']');
+  const textBeforeBracket = inputString.substring(0, openBracketIndex).trim();
+  const beginPhrase = 'Unable to switch to plan';
+  if (textBeforeBracket.startsWith(beginPhrase)) {
+    const [plan] =
+      textBeforeBracket.match(/OVHCLOUD_EDITION/) ??
+      textBeforeBracket.match(/STANDARD/) ??
+      [];
+    if (!plan) {
+      return null;
+    }
+    const contentInsideBrackets = inputString
+      .substring(openBracketIndex + 1, closeBracketIndex)
+      .trim();
+    if (contentInsideBrackets) {
+      const drivers = contentInsideBrackets.split(',');
+      if (drivers.length) {
+        return { drivers, plan };
+      }
+    }
+  }
+  return null;
```

Comment:
why not using regex to extract values ? it looks like it could be simplified a lot, i don't think you need bracket indexes etc